### PR TITLE
fix(segm): restore classifier loss weight

### LINKED ISSUE #1154: Instance Segmentation training loss not decreasing with with version > 1.7.0
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

# Problem:
With the rfdetr-1.8.1, instance segmentation training loss isn't decreasing smoothly. In some cases, loss is increasing. No matter which dataset version I use, or pre-trained weight (from my collection or from Roboflow) I start from, the training loss is struggling to go down with 1.8.1. However, this doesn't happen with 1.6.3. 
If I use pre-trained weights from Roboflow, the training loss goes down for several epochs until it starts to struggle. This never happened with 1.6.3. With release-1.8.1, the model is struggling to converge on even my old stable datasets.  

To eliminate the possibility that the root cause might be my dataset or pre-trained weights, I did the following experiments.

Two datasets:
- dataset 0.0.8 (stable)
- dataset 0.0.9 (new dataset)

Two pretrained weights:
- weights20260403 (stable)
- weights20260406 (stable)

RFDETR Version:
- 1.6.3 (stable)
- 1.8.1 (possible issues)

The experiments are named with the pattern:
`<training-date>_dataset<version>_<size><rfdetr-version>_weights<version>`

Anytime rfdetr 1.8.1 is used, the training loss does not decrease properly. Only focusing on the green and blue plots reveals that. The only change between green and blue plots is the rfdetr version number.  

Image  

Image  

Image  

Image   

### Environment

- rfder 1.8.1
- Google Colab
- A100 GPU
- Python 3.12.13
- torch 2.11.0+cu128
- pytorch-lightning 2.6.5

### Minimal Reproducible Example

```
from rfdetr import RFDETRSegMedium

aug_config = {
    "HorizontalFlip": {"p": 0.5},
    "VerticalFlip": {"p": 0.5},
    "Rotate": {"limit": 20, "p": 0.5},
    "Affine": {
        "scale": (0.8, 1.2),
        "translate_percent": (-0.1, 0.1),
        "p": 0.5,
    },
    "ColorJitter": {
        "brightness": 0.5,
        "contrast": 0.5,
        "saturation": 0.5,
        "hue": 0.5,
        "p": 0.5,
    },
}


if pretrain_weights:
    model = RFDETRSegMedium(pretrain_weights=pretrain_weights)
else:
    model = RFDETRSegMedium()


model.train(
    dataset_dir=dataset_dir,
    epochs=10,
    batch_size=16,
    grad_accum_steps=1,
    lr=1e-4,
    num_workers=2,
    device='cuda',
    output_dir=output_dir,
    segmentation_head=True,
    run_test=False,
    tensorboard=True,
    aug_config=aug_config,
    num_queries=100,
    num_select=100
)

```

### Additional

# Concerns:  
* We have grouped disjointed masks. In such cases, COCO-json contains multiple polygons under a single  "segmentation" key. [example comment in supervision](https://github.com/roboflow/supervision/issues/373#issuecomment-1831781340).
* We have masks with holes. [example](https://github.com/cvat-ai/cvat/issues/7030#issuecomment-3646989105)
* Maybe [release-1.7.0](https://github.com/roboflow/rf-detr/releases/tag/1.7.0) had some major changes that are impacting our training.

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Borda**
This pull request updates the default value for the segmentation classification loss coefficient and adds new tests to ensure correct behavior. The main change is lowering the default `cls_loss_coef` from 5.0 to 1.0, and tests are added to verify both the new default and explicit overrides.

**Configuration change:**

* The default value of `cls_loss_coef` in `SegmentationTrainConfig` was changed from 5.0 to 1.0 to match the effective value used in versions prior to 1.7 (`src/rfdetr/config.py`).

**Testing improvements:**

* Added a test to confirm that the default `cls_loss_coef` is now 1.0, ensuring backward compatibility with pre-1.7 behavior (`tests/training/test_args.py`).
* Added a test to verify that explicitly setting `cls_loss_coef` (e.g., to 5.0) is respected and forwarded correctly (`tests/training/test_args.py`).

resolves #1154

### CODE REVIEW

**Comment by Borda** [comment]
The behavior change came from the effective source of `args.cls_loss_coef`.

Before `faf8a228` (`refactor: BuilderArgs Protocol + ModelConfig/TrainConfig field deduplication (#841)`), `_namespace.py` forwarded:

```python
cls_loss_coef=mc.cls_loss_coef
```

For segmentation models, that meant the effective classifier loss weight was `ModelConfig.cls_loss_coef == 1.0`.

After `faf8a228`, namespace construction started using `TrainConfig.cls_loss_coef` when `ModelConfig.cls_loss_coef` was not explicitly set. For `SegmentationTrainConfig`, that default is `5.0`, so segmentation fine-tuning started weighting `loss_ce` 5x higher by default.

So this PR is restoring the pre-1.7 effective behavior for segmentation training: default `loss_ce` weight goes back to `1.0`, while explicit overrides such as `cls_loss_coef=5.0` are still respected.
